camel/messages/base.py

```diff
@@ -26,370 +26,306 @@
-        r"""Addition operator override for :obj:`BaseMessage`.
-
-        Args:
-            other (Any): The value to be added with.
+    def create_new_instance(self, content: Content) -> "BaseMessage":
+        return self.model_copy(update={"content": content})
 
-        Returns:
-            Union[BaseMessage, Any]: The result of the addition.
-        """
+    def __add__(self, other: Any) -> Union["BaseMessage", Any]:
         if isinstance(other, BaseMessage):
-            combined_content = self.content.__add__(other.content)
-        elif isinstance(other, str):
-            combined_content = self.content.__add__(other)
-        else:
-            raise TypeError(
-                f"Unsupported operand type(s) for +: '{type(self)}' and "
-                f"'{type(other)}'"
+            new_content = Content(
+                text=(self.content.text or []) + (other.content.text or []),
+                image_url=(self.content.image_url or [])
+                + (other.content.image_url or []),
+                video_url=(self.content.video_url or [])
+                + (other.content.video_url or []),
+                audio_url=(self.content.audio_url or [])
+                + (other.content.audio_url or []),
             )
-        return self.create_new_instance(combined_content)
+            return self.create_new_instance(new_content)
+        raise TypeError(
+            f"Unsupported operand type(s) for"
+            f" +: '{type(self)}' and '{type(other)}'"
+        )
 
     def __mul__(self, other: Any) -> Union["BaseMessage", Any]:
-        r"""Multiplication operator override for :obj:`BaseMessage`.
-
-        Args:
-            other (Any): The value to be multiplied with.
-
-        Returns:
-            Union[BaseMessage, Any]: The result of the multiplication.
-        """
         if isinstance(other, int):
-            multiplied_content = self.content.__mul__(other)
-            return self.create_new_instance(multiplied_content)
-        else:
-            raise TypeError(
-                f"Unsupported operand type(s) for *: '{type(self)}' and "
-                f"'{type(other)}'"
+            new_content = Content(
+                text=(self.content.text or []) * other,
+                image_url=(self.content.image_url or []) * other,
+                video_url=(self.content.video_url or []) * other,
+                audio_url=(self.content.audio_url or []) * other,
             )
+            return self.create_new_instance(new_content)
+        raise TypeError(
+            f"Unsupported operand type(s) for *:"
+            f" '{type(self)}' and '{type(other)}'"
+        )
 
     def __len__(self) -> int:
-        r"""Length operator override for :obj:`BaseMessage`.
-
-        Returns:
-            int: The length of the content.
-        """
-        return len(self.content)
+        return sum(
+            len(getattr(self.content, attr) or [])
```

Comment:
I believe that it would be better to have a method in the Content class so that we don't need to maintain the list of the names of the field in the Content class at multiple places.